test/test_misc.f90

```diff
@@ -334,8 +337,87 @@ program test_misc
   call shrink(array_of_reals, 2_i64)
   call test_array(itest)%assert(array_of_reals, [1, 2]*1.0_r64)
   
+  ! Hilbert transform tests (H)
+  ! fx1 -> function 1, fx2 -> function 2
+  ! - hfx1_even stores hilbert transform calculated for fx1, and for even number
+  ! - of points
+  xmin = -30.0
+  xmax = 30.0
+  n_even = 4000
+  n_odd = 4001
+  ! ind_even are indices to compare in case of even number of points
+  ! ind_odd are indices to compare in case of odd number of points
+  allocate(ind_even(6),ind_odd(5))
+  ind_even = [801, 1201, 1601, 2001, 2401, 2801]
+  ind_odd = [889, 1333, 1777, 2221, 2665]
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: even function, even points")
+  allocate(x_even(n_even), hfx1_even(n_even))
+  call linspace(x_even, xmin, xmax, n_even)
+  call Hilbert_transform(fx1_array(x_even), hfx1_even)
+  call test_array(itest)%assert(hfx1_even(ind_even), hfx1_array(x_even(ind_even)), &
+       tol = 2e-4_r64)
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: odd function, even points")
+  allocate(hfx2_even(n_even))
+  call Hilbert_transform(fx2_array(x_even), hfx2_even)
+  call test_array(itest)%assert(hfx2_even(ind_even), hfx2_array(x_even(ind_even)), &
+       tol = 1e-4_r64)
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: even function, odd points")
```

Comment:
Write the function instead of just saying even/odd function here and elsewhere.

Reply:
Do I write like - ```"Hilbert transform: function - sinx/(1+x^2) , odd points"```

Reply:
Something like: "Hilbert transform: f(x) = sin(x)/(1 + x^2), odd points" would be nice. Please use proper spacing around operators. Consult the style guide: https://github.com/nakib/elphbolt/blob/develop/STYLEGUIDE.org